blueprint/core/blueprint_EcmascriptEngine.cpp

```diff
@@ -252,8 +252,101 @@ namespace blueprint
             return result;
         }
 
+        struct TimeoutFunctionManager : private juce::MultiTimer
+        {
+            ~TimeoutFunctionManager() override {
+                for(const auto &[id, timer] : timeoutFunctions)
+                    stopTimer(id);
+            }
+
+            juce::var clearTimeout(const int id)
+            {
+                stopTimer(id);
+                const auto f = timeoutFunctions.find(id);
+                if(f != timeoutFunctions.cend())
+                    timeoutFunctions.erase(f);
+                return juce::var();
+            }
+
+            int newTimeout(const juce::var::NativeFunction f, const int timeoutMillis, const std::vector<juce::var>&& args, const bool repeats=false)
+            {
+                static int nextId = 0;
+                timeoutFunctions.emplace(nextId, TimeoutFunction(f, std::move(args), repeats));
+                startTimer(nextId, timeoutMillis);
+                return nextId++;
+            }
+
+            void timerCallback(int id) override
+            {
+                const auto f = timeoutFunctions.find(id);
+                if(f != timeoutFunctions.cend())
+                {
+                    const auto cb = f->second;
+                    std::invoke(cb.f, juce::var::NativeFunctionArgs(juce::var(), cb.args.data(), static_cast<int>(cb.args.size())));
+                    if(!cb.repeats)
+                    {
+                        stopTimer(id);
```

Comment:
Maybe here we can just do:

```
if (!cb.repeats) {
    (void) clearTimeout(id);
}
```

?

Reply:
We certainly can, but we've already retrieved an iterator above, which would be re-done by `clearTimeout`.  This is actually what I did first time around, but thought it might be the right side of DRY and performant this way.

Reply:
Ah ok, fair enough!